### PR TITLE
Allow setting queue for each job

### PR DESCRIPTION
**Pull request by devilankur18**
#39 Added queue support as per the latest code.

### CODE REVIEW

**Comment by pokonski** [comment]
Hey @devilankur18, thanks for the PR! Before we merge this can you clean up the unrelated changes you copied from here? https://github.com/aeris/gush/commit/c9f9d9237e53452b1c2b82f509da9a5e51275d7e

File: lib/gush/job.rb [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
-      :finished_at, :failed_at, :started_at, :enqueued_at, :payloads, :klass
+      :finished_at, :failed_at, :started_at, :enqueued_at, :payloads, :klass, :queue
     attr_reader :name, :output_payload, :params
 
     def initialize(opts = {})
       options = opts.dup
       assign_variables(options)
     end
 
+    def payload(clazz)
```
Comment (pokonski) [inline lib/gush/job.rb]: Please remove this method, it's not related and not needed

File: lib/gush/job.rb [hunk 2]
```diff
@@ -98,6 +105,9 @@ def has_no_dependencies?
     end
 
     private
+    def logger
```
Comment (pokonski) [inline lib/gush/job.rb]: Also not needed

File: lib/gush/worker.rb [hunk 3]
```diff
@@ -65,16 +65,46 @@ def elapsed(start)
...
         RedisMutex.with_lock("gush_enqueue_outgoing_jobs_#{workflow_id}-#{job_name}", sleep: 0.3, block: 2) do
           out = client.find_job(workflow_id, job_name)
-
           if out.ready_to_start?
+            puts "enqueing the job for execution #{out.to_s}"
             client.enqueue_job(workflow_id, out)
           end
+          out
+        end
+      rescue RedisMutex::LockError
```
Comment (pokonski) [inline lib/gush/worker.rb]: Unrelated to queuing

**Comment by devilankur18** [comment]
@pokonski updated !!

**Comment by pokonski** [comment]
Looks good! Can you also fix the one failing spec and I'll merge it :) Thanks!

**Comment by devilankur18** [comment]
@pokonski fixed !!

**Comment by pokonski** [comment]
@devilankur18 this is now released under version 2.0.0. Thanks again and sorry for the delay!

**Pull request merged**